src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java

```diff
@@ -0,0 +1,110 @@
+package dev.j3fftw.litexpansion.machine;
+
+import dev.j3fftw.litexpansion.Items;
+import dev.j3fftw.litexpansion.LiteXpansion;
+import io.github.thebusybiscuit.slimefun4.core.multiblocks.MultiBlockMachine;
+import io.github.thebusybiscuit.slimefun4.utils.SlimefunUtils;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.api.Slimefun;
+import me.mrCookieSlime.Slimefun.cscorelib2.inventory.InvUtils;
+import org.bukkit.Effect;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Sound;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.Dispenser;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RefinedSmeltery extends MultiBlockMachine {
+
+    public static final RecipeType REFINED_SMELTERY_RECIPE = new RecipeType(
+        new NamespacedKey(LiteXpansion.getInstance(), "refined_smeltery"),
+        Items.REFINED_SMELTERY,
+        "",
+        "&7Used to refine ingots"
+    );
+
+    private static final ItemStack stone_bricks = new ItemStack(Material.STONE_BRICKS);
+
+    public RefinedSmeltery() {
+        super(Items.LITEXPANSION, Items.REFINED_SMELTERY, new ItemStack[] {
+            null, new ItemStack(Material.STONE_BRICK_WALL), null,
+            stone_bricks, new ItemStack(Material.DISPENSER), stone_bricks,
+            null, new ItemStack(Material.FLINT_AND_STEEL), null
+        }, new ItemStack[0], BlockFace.DOWN);
+    }
+
+    @Override
+    public List<ItemStack> getDisplayRecipes() {
+        List<ItemStack> items = new ArrayList<>();
+
+        for (int i = 0; i < recipes.size() - 1; i += 2) {
+            items.add(recipes.get(i)[0]);
+            items.add(recipes.get(i + 1)[0]);
+        }
+
+        return items;
+    }
+
+    @Override
+    public void onInteract(Player p, Block b) {
+        Block dispBlock = b.getRelative(BlockFace.DOWN);
```

Comment:
Check if b is the stone brick wall

Reply:
it only works with stone brick wall
tested it just now